# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../glibc/arm-old-abi.patch
# Copyright (C) 2007 - 2008 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- glibc-2.6/ports/sysdeps/unix/sysv/linux/arm/nptl/sysdep-cancel.h.vanilla	2007-06-16 09:14:03.349441489 +0200
+++ glibc-2.6/ports/sysdeps/unix/sysv/linux/arm/nptl/sysdep-cancel.h	2007-06-16 09:17:42.269917056 +0200
@@ -126,3 +126,9 @@
 # define NO_CANCELLATION 1
 
 #endif
+
+#ifndef __ASSEMBLER__ 
+# define RTLD_SINGLE_THREAD_P \ 
+  __builtin_expect (THREAD_GETMEM (THREAD_SELF, \ 
+   header.multiple_threads) == 0, 1) 
+#endif 
